### PR TITLE
Add logic for remaining custom signal operators

### PR DESCRIPTION
**Pull request by kjelko**

### CODE REVIEW

File: test/unit/remote-config/condition-evaluator.spec.ts [hunk 1]
```diff
@@ -960,10 +960,17 @@ describe('ConditionEvaluator', () => {
               targetCustomSignalValues: ['foo', 'bar']
             }
           });
+          const condition2 = createNamedCondition('is_active', {
```
Comment (erikeldridge) [inline test/unit/remote-config/condition-evaluator.spec.ts:963]: Nit: naming the different test cases something more self-descriptive than "is_enabled" and "is_active" would be helpful.

**Review by erikeldridge — APPROVED** [review]

File: src/remote-config/condition-evaluator-internal.ts [hunk 2]
```diff
@@ -246,3 +271,23 @@ function compareStrings(
   const actual = String(actualValue);
   return targetValues.some((target) => predicateFn(target, actual));
 }
+
+// Compares numeric version strings against each other.
+function compareNumericVersions(version1String: string|number, version2String: string): number {
```
[the remark this answers is not in the document]
Comment (kjelko) [reply src/remote-config/condition-evaluator-internal.ts:276]: Done. This was a holdover from how things are named elsewhere :)

File: src/remote-config/condition-evaluator-internal.ts [hunk 3]
```diff
@@ -220,15 +220,40 @@ export class ConditionEvaluator {
...
     case CustomSignalOperator.STRING_CONTAINS_REGEX:
       return compareStrings(
         targetCustomSignalValues,
         actualCustomSignalValue,
         (target, actual) => new RegExp(target).test(actual),
       );
-     // TODO: add comparison logic for additional operators here.
+
+    case CustomSignalOperator.NUMERIC_LESS_THAN:
+      return Number(actualCustomSignalValue) < Number(targetCustomSignalValues[0]);
```
[the remark this answers is not in the document]
Comment (kjelko) [reply src/remote-config/condition-evaluator-internal.ts:233]: For numeric and version operators, only one value is allowed. This is true for the client implementation as well. Added some comments here to document that.

File: test/unit/remote-config/condition-evaluator.spec.ts [hunk 4]
```diff
@@ -1010,6 +1017,436 @@ describe('ConditionEvaluator', () => {
...
+              customSignalOperator: CustomSignalOperator.SEMANTIC_VERSION_LESS_EQUAL,
+              customSignalKey: 'user_prop',
+              targetCustomSignalValues: ['5.12.3']
+            }
+          });
+          const condition2 = createNamedCondition('is_active', {
+            customSignal: {
+              customSignalOperator: CustomSignalOperator.SEMANTIC_VERSION_LESS_EQUAL,
+              customSignalKey: 'user_prop',
+              targetCustomSignalValues: ['5.13']
```
[the remark this answers is not in the document]
Comment (kjelko) [reply test/unit/remote-config/condition-evaluator.spec.ts]: Updated a bunch of these tests. We've got pretty extensive coverage now

**Pull request merged**